src/functions/env.ts

```diff
@@ -0,0 +1,238 @@
+import * as fs from "fs";
+import * as path from "path";
+
+import { FirebaseError } from "../error";
+import { logger } from "../logger";
+
+const RESERVED_KEYS = [
+  // Cloud Functions for Firebase
+  "FIREBASE_CONFIG",
+  // Cloud Functions - old runtimes:
+  //   https://cloud.google.com/functions/docs/env-var#nodejs_8_python_37_and_go_111
+  "ENTRY_POINT",
+  "GCP_PROJECT",
+  "GCLOUD_PROJECT",
+  "GOOGLE_CLOUD_PROJECT",
+  "FUNCTION_TRIGGER_TYPE",
+  "FUNCTION_NAME",
+  "FUNCTION_MEMORY_MB",
+  "FUNCTION_TIMEOUT_SEC",
+  "FUNCTION_IDENTITY",
+  "FUNCTION_REGION",
+  // Cloud Functions - new runtimes:
+  //   https://cloud.google.com/functions/docs/env-var#newer_runtimes
+  "FUNCTION_TARGET",
+  "FUNCTION_SIGNATURE_TYPE",
+  "K_SERVICE",
+  "K_REVISION",
+  "PORT",
+  // Cloud Run:
+  //   https://cloud.google.com/run/docs/reference/container-contract#env-vars
+  "K_CONFIGURATION",
+];
+
+// Regex to capture key, value pair in a dotenv file.
+// Inspired by:
+//   https://github.com/bkeepers/dotenv/blob/master/lib/dotenv/parser.rb
+// prettier-ignore
+const LINE_RE = new RegExp(
+  "^" +                    // begin line
+  "\\s*" +                 //   leading whitespaces
+  "(\\w+)" +               //   key
+  "\\s*=\\s*" +            //   separator (=)
+  "(" +                    //   begin optional value
+  `\\s*'(?:\\'|[^'])*'|` + //     single quoted or
+  `\\s*"(?:\\"|[^"])*"|` + //     double quoted or
+  "[^\\#\\r\\n]+" +        //     unquoted
```

Comment:
What is `\#` in regexp? I can't find any reference to that. Do you mean a literal #?

Reply:
Huh I thought # had to be escaped, but the tests still pass without it. Thanks for simplifying this.